trin-bridge/src/utils.rs

```diff
@@ -0,0 +1,87 @@
+use crate::constants::BATCH_SIZE;
+use discv5::enr::{CombinedKey, EnrBuilder, NodeId};
+use std::ops::Range;
+use trin_utils::bytes::hex_encode;
+
+/// Splits a range into chunks of (at most) BATCH_SIZE,
+/// Used in "latest" mode, since most "chunks" will contain only a couple headers
+pub fn get_ranges(range: &Range<u64>) -> Vec<Range<u64>> {
+    let mut ranges = vec![];
+    let mut index = range.start;
+    while index < range.end {
+        let end = std::cmp::min(index + BATCH_SIZE, range.end);
+        let range = Range { start: index, end };
+        ranges.push(range);
+        index += BATCH_SIZE;
+    }
+    ranges
+}
+
+/// Generates a set of N private keys, with node ids that are equally spaced
+/// around the 256-bit keys space.
+pub fn generate_spaced_private_keys(count: u8) -> Vec<String> {
+    if count > 16 {
+        panic!("count must be less than 16");
+    }
+    let mut private_keys = vec![];
+    let (root_node_id, root_private_key) = random_node_id();
+    private_keys.push(hex_encode(root_private_key.encode()));
+    let mut root_prefix = root_node_id.raw()[0];
+
+    let mut prefixes: Vec<u8> = vec![];
+    for i in 1..count {
+        let prefix = root_prefix.wrapping_add(u8::MAX / i);
```

Comment:
Actually, I'm not sure you can guarantee a maximum distance, so we might just drop that requirement. Since it's an XOR, if the first key is really high, then the next key will be really low, and the XOR will be large. But I think the XOR distance metric should still keep a minimum distance working (empirically, the minimum distance check also seemed to not fail randomly on local tests).

Reply:
Thanks for digging! Yeah, i'm ok with dropping the max distance guarantee. The minimum distance guarantee is a more reliable metric imo